lib/rules/landmark-unique-matches.js

```diff
@@ -0,0 +1,39 @@
+const excludedParentsForHeaderFooterLandmarks = [
+	'article',
+	'aside',
+	'main',
+	'nav',
+	'section'
+].join(',');
+function isHeaderFooterLandmark(headerFooterElement) {
+	return !axe.commons.dom.findUp(
+		headerFooterElement,
+		excludedParentsForHeaderFooterLandmarks
+	);
+}
+
+function isLandmarkVirtual({ actualNode }) {
+	const landmarkRoles = axe.commons.aria.getRolesByType('landmark');
+	const role = axe.commons.aria.getRole(actualNode);
+	const tagName = actualNode.nodeName.toLowerCase();
+
+	if (tagName === 'header' || tagName === 'footer') {
+		return isHeaderFooterLandmark(actualNode);
+	}
+
+	if (tagName === 'section' || tagName === 'form') {
+		const label = axe.commons.aria.label(actualNode);
```

Comment:
This will be slow on large (deep) pages or with lots of landmarks or when both conditions hold. Need to pass the `virtualNode` into `isLandmarkVirtual` and then into here. Where we do not need the `virtualNode`, we can dereference it there.

Reply:
Replaced actual node usage with virtual node usage where there seemed to be virtual functions.